rfc/build-system/build-system.md

```diff
@@ -0,0 +1,711 @@
+the past year, they've each generated an echo 2.12.0, but those two 2.12.0s
+aren't related to one another in any way. Their git tags are slightly different
+(`v2.12.0` vs. `version-2.12.0`) and their artifacts are published to different
+Bintray repositories, so there are no conflicts. (Yes, we also have two
+different Bintray repositories.)
+
+## Timeline
+
+This build system will be implemented between the 1.22 release in mid-August and
+the following release (which will no longer be called 1.23) in mid-October.
+
+## Design
+
+### New Versioning Scheme and Release Schedule
+
+Spinnaker releases will switch to date-based versions, so instead of 1.23, we
+will call it 2020.4. The release schedule will change from every 8 weeks to
+every 3 months.
+
+Anecdotally, we believe that some users expect that there won't be any breaking
+changes between 1.20 and 1.21, perhaps confusing Spinnaker versions with
+[SemVer](https://semver.org/). In fact, we regularly introduce breaking changes
+in these releases.
+
+Additionally, this new version scheme will help users know how old their
+Spinnaker version is; we know many users are running very old versions of
+Spinnaker and this may goad them into upgrading more frequently.
+
+A third, lesser advantage of date-based versions is that it makes a clear break
+from microservice versions of the past. For example, there is already an Orca
+1.22.0 in one of the old version schemes.
+
+The release branches will be cut on the second Wednesday of the quarter, and the
+release will be targeted for the following Wednesday. (These dates were chosen
+to avoid major US holidays.) The time between releases will now be either 13 or
+14 weeks, depending on the calendar. The next releases will be:
+
+- 2020.4: branches cut 2020-10-14, release on 2020-10-21
+- 2021.1: branches cut 2021-01-13, release on 2021-01-20
+- 2021.2: branches cut 2021-04-14, release on 2021-04-21
+- 2021.3: branches cut 2021-07-14, release on 2021-07-21
+
+Slowing the release cadence reduces the release manager overhead somewhat,
+although release manager shifts are now longer and may have to be shared by
+multiple people. It also helps many of our users who find upgrading to be a lot
+of work by reducing how frequently they have to upgrade. It does mean it takes
+longer to get new features into their hands, however.
+
+**_Alternatives:_**
+
+- don't change the current Spinnaker version scheme (this decision really has
+  nothing to do with the rest of the RFC; the build system described here will
+  work with the existing Spinnaker versioning scheme too).
+- Use versions of YYYY.MM, where MM is the month. This makes it even more clear
+  when the release is from, but it looks a little weird with a patch release
+  (2020.07.3). It's also confusing since the first and last components of the
+  version increment by one, whereas the middle component changes by some other
+  interval.
+- Release more often. Release less often. Release every day?
```

Comment:
With longer release times, it seems like we have an opportunity for features to bake longer, which is nice. The downside is that it takes longer to get feedback from the community. Will there be any kind of `nightly` people can use to take things for a spin (heh)?